hbt/production/vbfjtag.py

```diff
@@ -0,0 +1,284 @@
+# coding: utf-8
+
+"""
+Producers for the VBFjTag score.
+See https://github.com/elviramartinv/VBFjtag/
+"""
+
+import law
+
+from columnflow.production import Producer, producer
+from columnflow.util import maybe_import, dev_sandbox, DotDict
+from columnflow.columnar_util import EMPTY_FLOAT, layout_ak_array, set_ak_column, full_like, flat_np_view
+from columnflow.types import Any
+
+from hbt.util import MET_COLUMN
+
+np = maybe_import("numpy")
+ak = maybe_import("awkward")
+
+logger = law.logger.get_logger(__name__)
+
+
+@producer(
+    uses={
+        "event", "channel_id",
+        "Jet.{pt,eta,phi,mass,jetId,btagDeepFlavB}",
+        MET_COLUMN("{pt,phi}"),
+    },
+    sandbox=dev_sandbox("bash::$HBT_BASE/sandboxes/venv_columnar_tf.sh"),
+)
+def vbfjtag(
+    self: Producer,
+    events: ak.Array,
+    vbfjet_mask: ak.Array,
+    lepton_pair: ak.Array,
+    is_hhbjet_mask: ak.Array,  # whether the jet was tagged by the hhbjet tagger
+    selected_fatjets: ak.Array,
+    **kwargs,
+) -> ak.Array:
+    """
+    Returns the VBFjTag score per passed jet.
+    Clean the scores from the selected fatjets and bjets before returning them.
+    """
+    # get a mask of events where there are at least two tau candidates and at least two jets
+    # and only get the scores for jets in these events
+
+    event_mask = (
+        (ak.num(lepton_pair, axis=1) >= 2) &
+        (ak.sum(vbfjet_mask, axis=1) >= 2)
+        # maybe just 3 since two of them would be the hhbjets if they are there?
+        # maybe add hhbjet_mask >=2, such that only these events are considered?
+    )
+
+    # ordering by decreasing eta then pt
+    f = 10**(np.ceil(np.log10(ak.max(events.Jet.pt))) + 2)
+    jet_sorting_key = events.Jet.eta * f + events.Jet.pt
+    jet_sorting_indices = ak.argsort(jet_sorting_key, axis=-1, ascending=False)
+
+    # back transformations for the saving of the scores
+    jet_unsorting_indices = ak.argsort(jet_sorting_indices)
+
+    # prepare objects
+    n_jets_max = 10
+    jets = events.Jet[jet_sorting_indices][vbfjet_mask[jet_sorting_indices]][event_mask][..., :n_jets_max]
```

Comment:
Just a detail, but `vbfjet_mask[jet_sorting_indices]` is used 5 (or so) times here and below. Can you save the sorted mask? This could also make some of the lines below more readable (and shorter).